### PR TITLE
Remove sine-wave from the pulsing color

### PR DESCRIPTION
**Pull request by dvdvideo1234**
Removing the sine makes it slightly faster

image

### CODE REVIEW

File: lua/entities/offset_hoverball/cl_init.lua [hunk 1]
```diff
@@ -168,39 +163,36 @@ function ENT:DrawMidArrows(PosX, PosY, PadY)
...
-
 -- Draws the rows of text for the mouseover UI. Turns out draw.SimpleText supports translating strings and checking text width.
-function ENT:DrawInfoContent(HBData, PosX, PosY, SizX, PadX, PadY, PadMid)
-    local Font = "OHBTipFontSmall"
-    local RowY = PosY
+function ENT:DrawInfoContent(HBData, PosX, PosY, SizX, PadX, PadY, PadM)
     local LongestL, LongestR = 0, 0
-
+    local Font, RowY = "OHBTipFontSmall", PosY
+    local RhbX, RhvX = (PosX + PadX), (PosX + (SizX - PadX))
```
Comment (dvdvideo1234) [inline lua/entities/offset_hoverball/cl_init.lua:170]: Depend only on the passed arguments. Use outside of the loop.

File: lua/entities/offset_hoverball/cl_init.lua [hunk 2]
```diff
@@ -267,8 +258,7 @@ hook.Add("HUDPaint", "OffsetHoverballs_MouseoverUI", function()
     local PoinX = HBPos:ToScreen().y - SizeP * 0.5
 
     -- Check to see if we will be drawing a header.
-    local IsDrawingHeader = false
-    if HBData[1] ~= "" then IsDrawingHeader = true end
```
Comment (dvdvideo1234) [inline lua/entities/offset_hoverball/cl_init.lua:271]: Boolean assignment is faster than switch condition

File: lua/entities/offset_hoverball/cl_init.lua [hunk 3]
```diff
@@ -278,15 +268,16 @@ hook.Add("HUDPaint", "OffsetHoverballs_MouseoverUI", function()
     local WidthHeader = 100
     
     -- Draws the info rows and returns the total width of the widest one.
-    WidthContent = LookingAt:DrawInfoContent(HBData, BoxX, BoxY+17, UIContainerWidth, PadX, PadY, PadMid)
+    WidthContent = LookingAt:DrawInfoContent(HBData, BoxX, BoxY+17, UIContainerWidth, PadX, PadY, PadM)
     
     -- Draw header.
     if IsDrawingHeader then
-        WidthHeader = LookingAt:DrawInfoTitle(MouseoverUI_HeaderKeys[tonumber(HBData[1]) or 0], BoxX, BoxY, UIContainerWidth, 40)
+        local Key = MouseoverUI_HeaderKeys[tonumber(HBData[1]) or 0]
```
Comment (dvdvideo1234) [inline lua/entities/offset_hoverball/cl_init.lua:275]: The line becomes quite long

File: lua/entities/offset_hoverball/cl_init.lua [hunk 4]
```diff
@@ -168,39 +163,36 @@ function ENT:DrawMidArrows(PosX, PosY, PadY)
...
-        local StrW1, StrH = draw.SimpleText(V, Font, hbx, RowY, CoOHBName, TEXT_ALIGN_LEFT, TEXT_ALIGN_CENTER)
+        local StrW1, StrH = draw.SimpleText(V, Font, RhbX, RowY, CoOHBName, TEXT_ALIGN_LEFT, TEXT_ALIGN_CENTER)
         LongestL = math.max(LongestL, StrW1)
         
         -- Right-side text shadow.
-        draw.DrawText(FormattedNum, Font, hvx+1, RowY-9, color_black, TEXT_ALIGN_RIGHT)
+        draw.DrawText(FormattedNum, Font, RhvX+1, RowY-9, color_black, TEXT_ALIGN_RIGHT)
 
-        local StrW2, _ = draw.SimpleText(FormattedNum, Font, hvx, RowY, CoOHBValue, TEXT_ALIGN_RIGHT, TEXT_ALIGN_CENTER)
+        local StrW2 = draw.SimpleText(FormattedNum, Font, RhvX, RowY, CoOHBValue, TEXT_ALIGN_RIGHT, TEXT_ALIGN_CENTER)
```
Comment (dvdvideo1234) [inline lua/entities/offset_hoverball/cl_init.lua:185]: We use only the first returned value anyway

File: lua/entities/offset_hoverball/cl_init.lua [hunk 5]
```diff
@@ -124,42 +120,41 @@ function ENT:DrawInfoPointy(PosX, PosY, SizX, SizY)
...
 
-    draw.RoundedBoxEx(8, PosX, PosY, SizX, SizY, CoOHBBack20, true, true, false, false)         -- Header Outline
-    draw.RoundedBoxEx(8, PosX+1, PosY+1, SizX-2, SizY, CoOHBBack70, true, true, false, false)   -- Header BG
+    draw.RoundedBoxEx(8, PosX, PosO, SizX, SizO, CoOHBBack20, true, true, false, false)         -- Header Outline
+    draw.RoundedBoxEx(8, PosX+1, PosO+1, SizX-2, SizO, CoOHBBack70, true, true, false, false)   -- Header BG
 
     draw.DrawText(StrT, "OHBTipFont", TxtX+1, TxtY-10, color_black, TEXT_ALIGN_CENTER)
     draw.SimpleText(StrT, "OHBTipFontGlow", TxtX, TxtY, CoHeaderPulse, TEXT_ALIGN_CENTER, TEXT_ALIGN_CENTER)
-    local TW, _ = draw.SimpleText(StrT, "OHBTipFont", TxtX, TxtY, CoOHBName, TEXT_ALIGN_CENTER, TEXT_ALIGN_CENTER)
+    local TW = draw.SimpleText(StrT, "OHBTipFont", TxtX, TxtY, CoOHBName, TEXT_ALIGN_CENTER, TEXT_ALIGN_CENTER)
```
Comment (dvdvideo1234) [inline lua/entities/offset_hoverball/cl_init.lua:153]: We use only the first returned value anyway

File: lua/entities/offset_hoverball/cl_init.lua [hunk 6]
```diff
@@ -124,42 +120,41 @@ function ENT:DrawInfoPointy(PosX, PosY, SizX, SizY)
...
 -- Draws the flashing title on the top of the mouseover UI. Is only called if there is actually a title.
 function ENT:DrawInfoTitle(StrT, PosX, PosY, SizX, SizY)
     local TxtX  = (PosX + (SizX / 2))
     local TxtY  = (PosY-12)
     local StrT  = tostring(StrT)
 
     -- Tweak position of background box.
-    PosY = PosY-30
-    SizY = SizY-8
+    local PosO = PosY-30
```
Comment (dvdvideo1234) [inline lua/entities/offset_hoverball/cl_init.lua:143]: Assign then to tweaks/offsets to preserve the scope

File: lua/entities/offset_hoverball/cl_init.lua [hunk 7]
```diff
@@ -124,42 +120,41 @@ function ENT:DrawInfoPointy(PosX, PosY, SizX, SizY)
...
 -- Base functionality for drawing the box container.
 function ENT:DrawInfoBox(PosX, PosY, SizX, SizY, TopCorners)
     draw.RoundedBoxEx(8, PosX  , PosY  , SizX  , SizY  , CoOHBBack20, TopCorners, TopCorners, true, true) -- Outline (Black)
     draw.RoundedBoxEx(8, PosX+1, PosY+1, SizX-2, SizY-2, CoOHBBack60, TopCorners, TopCorners, true, true) -- Inner background (Grey)
 end
 
-
-local function SineBetween(from, to, speed)
-    return math.Clamp(math.Remap(math.sin(CurTime()*speed)+0.005, -1, 1, from, to), from, to)
+local function RampBetween(from, to, speed)
```
Comment (dvdvideo1234) [inline lua/entities/offset_hoverball/cl_init.lua:129]: It is 10ms fatster ;)

**Comment by Sony-dev** [comment]
All good changes, thank you again!

**Pull request merged**